lib/bootstrap_form/helpers/bootstrap.rb

```diff
@@ -66,15 +66,33 @@ def custom_control(*args, &block)
         form_group_builder(name, options, &block)
       end
 
-      def prepend_and_append_input(options, &block)
-        options = options.extract!(:prepend, :append, :input_group_class)
+      ##
+      # Add prepend and append, if any, and error if any.
+      # If anything is added, the whole thing is wrapped in an input-group.
+      def prepend_and_append_input(name, options, &block)
+        control_error_help(name,
+                           options,
+                           prepend: options.delete(:prepend),
+                           append: options.delete(:append),
+                           &block)
+      end
+
+      ##
+      # Render a block, and add its error or help.
+      # Add prepend and append if provided, and wrap if they were, or if
+      # an input_group_class was provided.
+      def control_error_help(name, options, prepend: nil, append: nil, &block)
+        error_text = generate_help(name, options.delete(:help)).to_s
+        options = options.extract!(:input_group_class)
         input_group_class = ["input-group", options[:input_group_class]].compact.join(' ')
 
         input = capture(&block)
 
-        input = content_tag(:div, input_group_content(options[:prepend]), class: 'input-group-prepend') + input if options[:prepend]
-        input << content_tag(:div, input_group_content(options[:append]), class: 'input-group-append') if options[:append]
-        input = content_tag(:div, input, class: input_group_class) unless options.empty?
+        input = content_tag(:div, input_group_content(prepend), class: 'input-group-prepend') + input if prepend
+        input << content_tag(:div, input_group_content(append), class: 'input-group-append') if append
+        input << error_text
+        # FIXME: TBC The following isn't right yet. Wrap if there were errors. Maybe???
+        input = content_tag(:div, input, class: input_group_class) unless options.empty? && prepend.nil? && append.nil?
```

Comment:
Is this FIXME still valid? If so, could you clarify what additional cases need to be handled here?